firmware/Makefile

```diff
@@ -127,6 +127,20 @@ MAKEFLAGS += ${NUMJOBS}
 # Project, sources and paths
 #
 
+BOARDS_DIR = $(PROJECT_DIR)/config/boards
+
+include $(PROJECT_DIR)/config/boards/$(PROJECT_BOARD)/board.mk
+
+# If enabled, this option makes the build process faster by not compiling
+# modules not used in the current configuration.
+# without USE_SMART_BUILD all ChibiOS (including all drivers) are builded. And all drivers includes get included.
+ifeq ($(USE_SMART_BUILD),)
```

Comment:
let's split this PR into two smaller PRs? can we take care of makefile changes & bootloader maybe separately before flash changed?

the fact that we have failures here is an indication that we have too many changes in one PR